lib/typing/ext/parse.go

```diff
@@ -24,10 +25,25 @@ func ParseFromInterface(val any, additionalDateFormats []string) (*ExtendedTime,
 	return extendedTime, nil
 }
 
-// ParseTimeExactMatch is a wrapper around time.Parse() and will return an extra boolean to indicate if it was an exact match or not.
+// ParseTimeExactMatch - This function is the same as `ParseTimeExactMatchLegacy` with the only exception that it'll return if it was not an exact match
+func ParseTimeExactMatch(layout, potentialDateTimeString string) (time.Time, error) {
+	ts, err := time.Parse(layout, potentialDateTimeString)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	if ts.Format(layout) != potentialDateTimeString {
```

Comment:
In what case would a formatted value not be the same as the value that was parsed? Is this to ensure that we are only matching time strings that have the same microsecond precision as the layout?

Reply:
Replied via email but the format got messed up.

Yes, it's to ensure that precision did not get truncated.

Reply:
For example:
1. potentialDateTimeString = 15:04:05.000000Z
2. layout = 15:04:05Z

It will parse correctly. However, if we went with this layout, we would
lose microsecond precision.

Reply:
Gotcha, could we add a comment here to that effect?

Reply:
Yup